projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,432 @@
+
+## 3.2 `htr_createToken`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will call `prepareCreateNewToken` and display the assembled transaction for user validation (UX is yet to be discussed).
+
+If the user accepts the transaction, we will, just like the `htr_sendTx` method, create a `SendTransaction` model and execute the send transaction. Otherwise, we will answer with a `REQUEST_REJECTED` error
+
+## 3.3 `htr_getUtxos`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will call the `getUtxos` method to retrieve `utxos` that fulfill the requested filters.
+
+Before returning, we will display a confirmation screen (UX yet to be discussed) asking for user confirmation by displaying the `utxos` that will be returned to the `dApp`.
+
+If the user accepts the transaction, we will reply with the array of `utxos`. Otherwise, we will answer with a `REQUEST_REJECTED` error
+
+#### Future possibilities
+
+We might display "checkboxes" before each utxos so the user can choose exactly which ones he want to display
+
+## 3.4 `htr_signWithAddress`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will display the `message` and the `address` to the user, asking for his confirmation. If he rejects the modal, we will respond with an `INVALID_PARAMETERS` error code.
+
+If he accepts, we will call the `signMessageWithAddress` method from the lib and answer the request
+
+## 3.5 `htr_getBalance`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will display the `message` and the `address` to the user, asking for his confirmation. If he rejects the modal, we will respond with an `INVALID_PARAMETERS` error code.
+
+If he accepts, we will call the `getBalance` method from the lib and answer the request
+
+## 3.6 `htr_getConnectedNetwork`
+
+This method should not require explicit approval of the wallet given that the wallet and the dApp are already connected.
+
+It will return the wallet's current connected network, as defined in the reference-level section
```

Comment:
Should we notify users that the app is requesting information?

Reply:
I don't think we should in this specific case, he already approved the dApp connection, fetching the network seems harmless to me